### ISSUE TITLE
Proxy Setup

### ISSUE THREAD

**Issue by fedex03** [instabot-py/instabot.py#1542]
Check this or we will delete your issue. (fill in the checkbox with an X like so: [x])
- [x] I have searched for other issues with the same problem. (including #1153)

## Hosting
- [x] Are you hosting the bot on a VPS or other Cloud hosting?
- [ ] If so; have you tried running the code on your local machine?

#### Select one:
- [x] Bug
- [ ] Feature Request
- [x] Technical Help
- [ ] Other

### Operating System? (include version)
- [ ] MAC
- [ ] Windows
- [x] Linux
- [ ] Android

## Python Version?
- [x] 2.x
- [ ] 3.x

## Description of your issue

```
I there,
I cannot use my proxy (I bought a proxy from **https://proxy6.net**). This is a HTTPS proxy and I add the proxy settings into the configuration but I get login error. 
I tried the proxy on my PC and it works.

Did someone try the bot with proxy?

Thanks in advance.

```

## Config

```
    login="user",
    password="pwd",
    start_at_h=6,
    start_at_m=10,
    end_at_h=23,
    end_at_m=55,
    like_per_day=1000,
    media_min_like=20,
    comments_per_day=20,
    tag_list=['xxx', 'xxx', 'xxx', 'xxx', 'xxx' ],
    tag_blacklist=[],
    user_blacklist={},
    max_like_for_one_tag=150,
    follow_per_day=0,
    follow_time=5 * 60 * 60,
    unfollow_per_day=300,
    unfollow_break_min=1,
    unfollow_break_max=5,
    log_mod=0,
    proxy='user:password@host:port',
```

**Comment by senatorlila** [comment]
Same issue!

**Comment by fedex03** [comment]
I found a way to use proxies.

In a nutshell to use proxies with authentication with the Request module is not possible to include the authentication password and username into the proxy address.

The authentication parameters should be passed via **HTTPProxyAuth**

To test this I wrote this simple python script to test the authentication:
`import requests
s = requests.Session()

proxies = {
  "http": "http://ip:port",
  "https": "https://ip:port"
}

auth = HTTPProxyAuth("user", "pwd")
ext_ip = s.get('http://checkip.dyndns.org', proxies=proxies, auth=auth)
print ext_ip.text`

Basically the authentication parameter should be passed to each s.get function inside the bot script.

I cannot found a method to set the authentication parameter globally. But I'm not a python master, maybe someone could found a solution.

BR,
Federico

**Comment by marianoarga** [comment]
@fedex03 did you figure it out how to fix this issue with proxy authentication?

**Comment by caioladislau** [comment]
Same issue!

**Comment by DarkSuniuM** [comment]
Closed all the issues till 2019-01-09, We updated the repo, Please create a new issue if you need! Thanks
